tests/Kiota.Builder.Tests/Refiners/TypeScriptLanguageRefinerTests.cs

```diff
@@ -51,34 +53,27 @@ public async Task AddsQueryParameterMapperMethod()
     [Fact]
     public async Task AddStaticMethodsUsingsForDeserializer()
     {
-        var model = graphNS.AddClass(new CodeClass
-        {
-            Name = "somemodel",
-            Kind = CodeClassKind.Model,
-            IsErrorDefinition = true,
-        }).First();
+        var model = TestHelper.CreateModelClass("Model");
+        graphNS.AddClass(model);
+
+        var subNs = graphNS.AddNamespace($"{graphNS.Name}.subns");
+
+        var propertyModel = TestHelper.CreateModelClass("PropertyModel");
+        subNs.AddClass(propertyModel);
 
         model.AddMethod(new CodeMethod
         {
-            Name = "Deserialize",
-            Kind = CodeMethodKind.Deserializer,
+            Name = "factory",
+            Kind = CodeMethodKind.Factory,
```

Comment:
I don't think the kind and name changes are required here anymore, I think that change came from a previous state of the code that's no longer require (I can't find any trace in the refiner of a method changing the kind from deserializer to factory)